src/streamlink/session.py

```diff
@@ -387,13 +392,40 @@ def resolve_url(self, url, follow_redirect=True):
         :param follow_redirect: follow redirects
 
         """
+        default_plugin_url_copy = url
+        default_plugin = self.get_option("default-plugin")
         url = update_scheme("http://", url)
 
         available_plugins = []
+
         for name, plugin in self.plugins.items():
             if plugin.can_handle_url(url):
                 available_plugins.append(plugin)
 
+        if len(available_plugins) == 0 and default_plugin is not None:
+            plugin_found = False
+
+            for name, plugin in self.plugins.items():
+                if plugin.__name__.lower() == default_plugin.lower():
+                    plugin_found = True
+                    for string in plugin.get_urls():
+                        if plugin.can_handle_url(string + default_plugin_url_copy):
```

Comment:
This is what I meant with the URL schemes. You can't expect every plugin/website to have their URL formatted like this where the channel name or video ID or something else gets added to the end of the defined plugin URLs. You need to use a URL template that replaces a `{value}` variable for example.

**Example:**

```python
# string = "https://twitch.tv/{value}"
# string = "https://youtube.com/watch?v={value}"
# string = "https://some.streaming.website/{value}/live"
string.format(value=default_plugin_url_copy)
```

This should also be done only once (you are currently concatenating the URL parts 3 times).